tendrl/commons/objects/cluster/atoms/import_cluster/__init__.py

```diff
@@ -239,10 +232,41 @@ def run(self):
                                 job_id=child_job_id
                             ).load()
                             if child_job.status not in ["finished", "failed"]:
+                                if child_job.status in ["new", ""]:
+                                    node_id = child_job.payload.get(
+                                        "node_id", ""
+                                    )
+                                    node_obj = NS.tendrl.objects.NodeContext(
+                                        node_id = node_id
+                                    ).load()
+                                    logger.log(
+                                        "error",
+                                        NS.publisher_id,
+                                        {"message": "Import child job %s is "
+                                         "not yet picked by %s, Either node is"
+                                         " down or tendrl-node-agent service "
+                                         "is not running" % (
+                                             child_job.job_id,
+                                             node_obj.fqdn
+                                         )},
+                                        job_id=self.parameters['job_id'],
+                                        flow_id=self.parameters['flow_id']
+                                    )
                                 child_job.status = "failed"
                                 child_job.save()
+
+                        logger.log(
+                            "error",
+                            NS.publisher_id,
+                            {"message": "Import jobs on cluster(%s) not yet "
+                             "complete on all nodes(%s). Timing out." %
+                             (_cluster.short_name, str(node_list))},
+                            job_id=self.parameters['job_id'],
+                            flow_id=self.parameters['flow_id']
+                        )
+
                         return False
-                    time.sleep(10)
+                    time.sleep(5)
                     completed = True
```

Comment:
10 sec per sleep is too much for import fail, we did all calculation for 5 sec only, it will work for a large number of volumes. because we are setting timeout based on a number of volumes. Also, we reduced gluster-integration sync time so we don't need 10 sec sleep.

Reply:
Actually as per the calculation it should be sleep for 5 sec only so that 96*5 = 480 sec = 8 mins

Reply:
ack